File: iOS ObjectiveC Example/iOS ObjectiveC Example/HelloClient.m

#import "HelloClient.h"

@interface HelloClient()
@property (nonatomic, strong) NSString *baseUrl;
@end

@implementation HelloClient

- (id)initWithBaseUrl:(NSString *)url {
  if (self = [super init]) {
    self.baseUrl = url;
  }
  return self;
}

- (NSString *)sayHello {
  NSString *url = [NSString stringWithFormat:@"%@/%@", self.baseUrl, @"sayHello"];
  NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:url]
                                                         cachePolicy:NSURLRequestReloadIgnoringLocalAndRemoteCacheData
                                                     timeoutInterval:10];
  
  [request setHTTPMethod: @"GET"];
  NSError *requestError;
  NSURLResponse *urlResponse = nil;
  
  NSData *response = [NSURLConnection sendSynchronousRequest:request returningResponse:&urlResponse error:&requestError];
  
  NSString *requestReply = [[NSString alloc] initWithBytes:[response bytes] length:[response length] encoding:NSASCIIStringEncoding];
  NSLog(@"requestReply: %@", requestReply);
  
  return requestReply;
}

- (NSString *)findFriendsByAgeAndChild {
  NSString *url = [NSString stringWithFormat:@"%@/%@", self.baseUrl, @"friends?age=30&child=Mary"];
  NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:url]
                                                         cachePolicy:NSURLRequestReloadIgnoringLocalAndRemoteCacheData
                                                     timeoutInterval:10];
  
  [request setHTTPMethod: @"GET"];
  NSError *requestError;
  NSURLResponse *urlResponse = nil;
  
  NSData *response = [NSURLConnection sendSynchronousRequest:request returningResponse:&urlResponse error:&requestError];
  
  NSString *requestReply = [[NSString alloc] initWithBytes:[response bytes] length:[response length] encoding:NSASCIIStringEncoding];
  NSLog(@"requestReply: %@", requestReply);
  
  return requestReply;
}

- (void) unfriend:(void (^)(NSString *response))success failure:(void (^)(NSInteger errorCode))failure {
  NSString *url = [NSString stringWithFormat:@"%@/%@", self.baseUrl, @"unfriendMe"];
  NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:url]
                                                         cachePolicy:NSURLRequestReloadIgnoringLocalAndRemoteCacheData
                                                     timeoutInterval:10];
  
  [request setHTTPMethod: @"PUT"];
  NSError *requestError;
  NSHTTPURLResponse *urlResponse = nil;
  
  NSData *response = [NSURLConnection sendSynchronousRequest:request returningResponse:&urlResponse error:&requestError];
  
  if(urlResponse.statusCode > 299) {
    failure(urlResponse.statusCode);
  } else {
    NSString *requestReply = [[NSString alloc] initWithBytes:[response bytes] length:[response length] encoding:NSASCIIStringEncoding];
    NSLog(@"requestReply: %@", requestReply);
    success(requestReply);
  }
}

@end

File: iOS Swift Example/iOS Swift Example/HelloClient.swift
import Foundation
import Alamofire

public class HelloClient {
    private let baseUrl: String

  public init(baseUrl : String) {
    self.baseUrl = baseUrl
  }

  public func sayHello(helloResponse: (String) -> Void) {
    Alamofire.request(.GET, "\(baseUrl)/sayHello")
    .responseJSON {
      (request, response, json, error) in
      println(request)
      println(response)
      println(error)
      if let jsonResult = json as? Dictionary<String, AnyObject> {
        helloResponse(jsonResult["reply"] as String)
      }
    }
  }

  public func findFriendsByAgeAndChild(friendsResponse: (Array<String>) -> Void) {
    Alamofire.request(.GET, "\(baseUrl)/friends", parameters: [ "age" : "30", "child" : "Mary" ] )
    .responseJSON { (request, response, json, error) in
      println(request)
      println(response)
      println(error)
      if let jsonResult = json as? Dictionary<String, AnyObject> {
        friendsResponse(jsonResult["friends"] as Array)
      }
    }
  }
  
  public func unfriendMe(successResponse: (Dictionary<String, String>) -> Void, errorResponse: (Int) -> Void) {
    Alamofire.request(.PUT, "\(baseUrl)/unfriendMe" )
      .responseJSON { (request, response, json, error) in
        println(request)
        println(response)
        println(error)
        if let errorVal = error {
          errorResponse(response!.statusCode)
        }
        if let jsonResult = json as? Dictionary<String, AnyObject> {
          successResponse(jsonResult as Dictionary<String, String>)
        }
    }
  }
}